kube_client.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"errors"
+	"github.com/JaSei/pathutil-go"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/typed/extensions/v1beta1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func IngressClientAllNamespaces(clientset *kubernetes.Clientset) v1beta1.IngressInterface {
+	return IngressClient(metav1.NamespaceAll, clientset)
+}
+
+func IngressClient(namespace string, clientset *kubernetes.Clientset) (ingresses v1beta1.IngressInterface) {
+	ingresses = clientset.ExtensionsV1beta1().Ingresses(namespace)
```

Comment:
I use (in own similar project) for listing ingresses `clientset.Extensions().Ingresses(namespace)` it's probably more stable API?

Reply:
it doesn't matter both return `v1beta1.ExtensionsV1beta1Interface`